This patch fixes a problem with diagonal fonts being displayed in mozilla-devel
with Xft support enabled.  It was obtained from:

http://www.xfree86.org/pipermail/fonts/2002-September/002161.html

This is already in freetype2 CVS, and should not be needed when the next
release of freetype2 comes out.
--- src/base/ftoutln.c.orig	Thu Oct 24 14:32:25 2002
+++ src/base/ftoutln.c	Thu Oct 24 14:33:09 2002
@@ -628,9 +628,9 @@
       return;
 
     xz = FT_MulFix( vector->x, matrix->xx ) +
-         FT_MulFix( vector->y, matrix->yx );
+         FT_MulFix( vector->y, matrix->xy );
 
-    yz = FT_MulFix( vector->x, matrix->xy ) +
+    yz = FT_MulFix( vector->x, matrix->yx ) +
          FT_MulFix( vector->y, matrix->yy );
 
     vector->x = xz;
--- src/truetype/ttgload.c.orig	Thu Oct 24 14:33:38 2002
+++ src/truetype/ttgload.c	Thu Oct 24 14:34:09 2002
@@ -544,8 +544,8 @@
       else if ( subglyph->flags & WE_HAVE_A_2X2 )
       {
         xx = (FT_Fixed)FT_GET_SHORT() << 2;
-        xy = (FT_Fixed)FT_GET_SHORT() << 2;
         yx = (FT_Fixed)FT_GET_SHORT() << 2;
+	xy = (FT_Fixed)FT_GET_SHORT() << 2;
         yy = (FT_Fixed)FT_GET_SHORT() << 2;
       }
 
